### PR TITLE
Update changelog to add v0.6.0-rc1

### PR DESCRIPTION
**Pull request by youngnick**
Signed-off-by: Nick Young <nick@isovalent.com>

What this PR does / why we need it:
This PR contains the changelog for v0.6.0-rc1. By approving this PR, you are also approving the release of v0.6.0-rc1 based on the current state of the main branch of this project and the changelog included with this PR.

/cc @bowei @shaneutt @robscott 

/kind documentation

### CODE REVIEW

**Comment by youngnick** [comment]
This one is an update of #1541 with some extra details. Please check it out, and once it's good to go, I can cut v0.6.0-rc1.

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +19,126 @@
 - [v0.1.0-rc2](#v010-rc2)
 - [v0.1.0-rc1](#v010-rc1)
 
+# v0.6.0-rc1
+
+## Major Changes
+
+### ReferenceGrant moves to `v1beta1`, ReferencePolicy removed
+
+With more implementations now supporting ReferenceGrant, and more conformance coverage of the object, we've moved ReferenceGrant to `v1beta1` in this release. Note that moving to beta also moves the object to the Standard channel (it was Experimental previously).
```
Comment (shaneutt) [inline CHANGELOG.md:28]:
```suggestion
With more implementations now supporting ReferenceGrant (and more conformance coverage of the resource) we've moved ReferenceGrant to `v1beta1` in this release. **Note** that moving to beta also moves the object to the Standard channel (it was Experimental previously).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +19,126 @@
...
+
+We've also removed the already-deprecated ReferencePolicy resource, so please move over to the shiny new ReferenceGrant, which has all the same features.
+
```
Comment (shaneutt) [inline CHANGELOG.md:31]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,6 +19,126 @@
...
+
+- Promotes ReferenceGrant to the v1beta1 API and the standard release channel
+  (#1455, @nathancoleman)
+- ReferencePolicy has been removed from the API in favor of ReferenceGrant.
+  (#1406, @robscott)
+
```
Comment (shaneutt) [inline CHANGELOG.md:37]:
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -18,6 +19,126 @@
...
+### Introduce GRPCRoute
+
+The `GRPCRoute` resource has been introduced in order to simplify the routing of GRPC requests.
+Its design is described in [GEP-1016](https://gateway-api.sigs.k8s.io/geps/gep-1016/).
+As it is a new resource, it is introduced in the experimental channel.
+
+Thanks to @gnossen for pushing this ahead.
+
+- Introduce GRPCRoute resource. (#1115, @gnossen)
+
```
Comment (shaneutt) [inline CHANGELOG.md:48]:
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -18,6 +19,126 @@
...
+* `Detached` condition deprecated
+* Core Conditions now `Accepted`, `Programmed`, `ResolvedRefs`, and `Conflicted`
+* Moves to Extended: `Ready`
+
+GatewayClass and HTTPRoute:
+* No changes (both already used `Accepted`).
+
+The purpose of these changes is to make the status flows more consistent across objects, and to provide a clear pattern for new objects as we evolve the API.
+
+**This change will require updates for implementations to be able to pass conformance tests**.
```
Comment (shaneutt) [inline CHANGELOG.md:73]: Github supports notes and warnings with something like:

```suggestion
> **Note**: This change will require updates for implementations to be able to pass conformance tests.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -18,6 +19,126 @@
...
+  GatewayConditionScheduled and GatewayReasonScheduled (#1447, @mikemorris)
+- A `Pending` Reason has been added to Conditions to harmonize the unreconciled
+  state across objects. GatewayClass, Gateway, and Route `Accepted` Conditions
+  have been updated. (#1453, @youngnick)
+- Added "Programmed" Gateway and Listener conditions, moved "Ready" to extended
+  conformance (#1499, @LCaparelli)
+- Status definitions have been updated across the API. (#1383, @youngnick)
+- Adds `ListenerConditionAccepted` and `ListenerReasonAccepted` Deprecates
+  `ListenerConditionDetached` and `ListenerReasonAttached` (#1446, @mikemorris)
+
```
Comment (shaneutt) [inline CHANGELOG.md:85]:
```suggestion
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -18,6 +19,126 @@
...
+- GatewayClass, Gateway, and HTTPRoute are now only supported with the v1beta1
+  version of the API. The v1alpha2 API versions of these resources will be fully
+  removed in a future release. (#1348, @robscott)
+- V1alpha2 has been deprecated for Gateway, GatewayClass, and HTTPRoute. This
+  version of the API will be removed from CRDs in a future release.
+  Implementations should update to use v1beta1 for these resources as soon as
+  possible. (#1405, @robscott)
+
+
```
Comment (shaneutt) [inline CHANGELOG.md:99]:
```suggestion
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -18,6 +19,126 @@
...
+
+### API Changes
+
```
Comment (shaneutt) [inline CHANGELOG.md:102]:
```suggestion
```

**Review by shaneutt — APPROVED** [review]
Approved, but I would like to let Rob give the final LGTM.

File: CHANGELOG.md [hunk 9]
```diff
@@ -18,6 +19,119 @@
 - [v0.1.0-rc2](#v010-rc2)
 - [v0.1.0-rc1](#v010-rc1)
 
+# v0.6.0-rc1
+
+## Major Changes
+
+### ReferenceGrant moves to `v1beta1`, ReferencePolicy removed
+
+With more implementations now supporting ReferenceGrant  (and more conformance coverage of the resource), we've moved ReferenceGrant to `v1beta1` in this release. **Note** that moving to beta also moves the object to the Standard channel (it was Experimental previously).
```
Comment (robscott) [inline CHANGELOG.md:28]: Tiny nit:
```suggestion
With more implementations now supporting ReferenceGrant (and more conformance coverage of the resource), we've moved ReferenceGrant to `v1beta1` in this release. **Note** that moving to beta also moves the object to the Standard channel (it was Experimental previously).
```

**Review by robscott — COMMENTED** [review]
Thanks for getting this together @youngnick!

**Comment by robscott** [comment]
Thanks @youngnick!

/lgtm

**Pull request merged**